frontend/app/components/session-timeout.tsx

```diff
@@ -0,0 +1,67 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { useTranslation } from 'react-i18next';
+import { type IIdleTimerProps, useIdleTimer } from 'react-idle-timer';
+
+import { Button } from './buttons';
+import { Dialog, DialogContent, DialogFooter, DialogHeader, DialogTitle } from '~/components/dialog';
+import { getTypedI18nNamespaces } from '~/utils/locale-utils';
+
+const i18nNamespaces = getTypedI18nNamespaces('gcweb');
+
+export interface SessionTimeoutProps extends Pick<IIdleTimerProps, 'promptBeforeIdle'>, Pick<IIdleTimerProps, 'timeout'> {}
+
+const SessionTimeout = ({ promptBeforeIdle, timeout }: SessionTimeoutProps) => {
+  const { t } = useTranslation(i18nNamespaces);
+  const [modalOpen, setModalOpen] = useState(false);
+  const [timeRemaining, setTimeRemaining] = useState('');
+
+  const handleOnIdle = () => {
+    window.location.href = '/auth/logout';
+  };
+
+  const handleOnIdleContinueSession = () => {
+    setModalOpen(false);
+    reset();
+  };
```

Comment:
I would move this down since the `reset()` function is initialized in the block of code before..

so:

``` typescript
const { reset, getRemainingTime } = useIdleTimer({ .. })
// then
const handleOnIdleContinueSession = () => { .. }
```